### PR TITLE
EVG-6693 Normalize branch or tag name to git hash for patch-file command

### PR DESCRIPTION
**Pull request by bsamek**

### CODE REVIEW

File: units/patch_intent.go [hunk 1]
```diff
@@ -345,12 +345,17 @@ func (j *patchIntentProcessor) buildCliPatchDoc(ctx context.Context, patchDoc *p
...
-	_, err = thirdparty.GetCommitEvent(ctx, githubOauthToken, projectRef.Owner,
+	commit, err := thirdparty.GetCommitEvent(ctx, githubOauthToken, projectRef.Owner,
 		projectRef.Repo, patchDoc.Githash)
 	if err != nil {
 		return errors.Wrapf(err, "could not find base revision '%s' for project '%s'",
 			patchDoc.Githash, projectRef.Identifier)
 	}
+	// With `evergreen patch-file`, a user can pass a branch name or tag instead of a hash. We
+	// must normalize this to a hash before storing the patch doc.
+	if commit.SHA != nil && patchDoc.Githash != *commit.SHA {
```
Comment (ybrill) [inline units/patch_intent.go]: I think if the `patch.Githash` is not a SHA GitHub's API will not return the commit.
```
> curl -H "Authorization: token MY_TOKEN" https://api.github.com/repos/evergreen-ci/evergreen/git/commits/master
{
  "message": "Not Found",
  "documentation_url": "https://developer.github.com/v3/git/commits/#get-a-commit"
}
```

Comment (bsamek) [reply units/patch_intent.go]: I got a response from `curl -i -H 'Authorization: token TOKEN' https://api.github.com/repos/evergreen-ci/evergreen/commits/master`. I think you have an extra "git". https://developer.github.com/v3/repos/commits/#get-a-single-commit

Comment (ybrill) [reply units/patch_intent.go]: Yes, you're correct.

File: thirdparty/github.go [hunk 2]
```diff
@@ -263,7 +263,8 @@ func GetCommitEvent(ctx context.Context, oauthToken, repoOwner, repo, githash st
 		"operation": "github api query",
 		"size":      resp.ContentLength,
 		"status":    resp.Status,
-		"commit":    githash,
+		"query":     githash,
+		"commit":    commit,
```
Comment (ybrill) [inline thirdparty/github.go]: do we want to log the entire commit object?

Comment (bsamek) [reply thirdparty/github.go]: I don't think so. I just wanted to be able to validate the query and response SHA.

Comment (ybrill) [reply thirdparty/github.go]: isn't this logging the entire object, though?

Comment (bsamek) [reply thirdparty/github.go]: Oh, sorry, you're right. I thought you were asking the opposite. Fixed.

File: thirdparty/github.go [hunk 3]
```diff
@@ -259,13 +259,17 @@ func GetCommitEvent(ctx context.Context, oauthToken, repoOwner, repo, githash st
...
 		"operation": "github api query",
 		"size":      resp.ContentLength,
 		"status":    resp.Status,
-		"commit":    githash,
+		"query":     githash,
 		"repo":      repoOwner + "/" + repo,
-	})
+	}
+	if commit != nil {
+		msg["commit"] = *commit.SHA
```
Comment (ybrill) [inline thirdparty/github.go]: maybe it would be better to also validate the SHA isn't nil?

Comment (bsamek) [reply thirdparty/github.go]: True. Fixed.

**Pull request merged**